test/pobox_give_away_SUITE.erl

```diff
@@ -0,0 +1,88 @@
+-module(pobox_give_away_SUITE).
+-include_lib("common_test/include/ct.hrl").
+-compile(export_all).
+
+all() -> [
+  owner_gives_away_ownership,
+  owner_gives_away_ownership_with_data,
+  owner_gives_away_ownership_to_dead_process,
+  foreign_process_tries_to_gives_away_ownership,
+  give_away_does_not_change_heir
+].
+
+owner_gives_away_ownership(_Config) ->
+    {ok, Box} = pobox:start_link([]),
+    Self = self(),
+    Func = fun() ->
+      Fn = fun(F) -> receive X -> Self ! X end, F(F) end,
+      Fn(Fn)
+    end,
+    H = erlang:spawn(Func),
+    true = pobox:give_away(Box, H, 5000),
+    true = receive
+        {pobox_transfer, Box, Self, undefined, give_away} -> true;
+        Other -> Other
+    after
+      5000 -> false
+    end.
+
+owner_gives_away_ownership_with_data(_Config) ->
+    {ok, Box} = pobox:start_link([]),
+    Self = self(),
+    Func = fun() ->
+        Fn = fun(F) -> receive X -> Self ! X end, F(F) end,
+        Fn(Fn)
+    end,
+    H = erlang:spawn(Func),
+    true = pobox:give_away(Box, H, Ref = make_ref(), 5000),
+    true = receive
+        {pobox_transfer, Box, Self, Ref, give_away} -> true;
+        _Other -> false
+    after
+        5000 -> false
+    end.
+
+owner_gives_away_ownership_to_dead_process(_Config) ->
+    {ok, Box} = pobox:start_link([]),
+    Self = self(),
+    Func = fun() ->
+        Fn = fun(F) -> receive X -> Self ! X end, F(F) end,
+        Fn(Fn)
+    end,
+    H = erlang:spawn(Func),
+    erlang:exit(H, kill),
+    timer:sleep(100),
+    false = pobox:give_away(Box, H, 5000),
+    true = receive
+         _ -> false
+    after
+        100 -> true
+    end.
+
+
+foreign_process_tries_to_gives_away_ownership(_Config) ->
+    {ok, Box} = pobox:start_link([]),
+    Self = self(),
+    erlang:spawn(fun() -> Self ! pobox:give_away(Box, self(), 5000) end),
+    true = receive
+        true -> false;
+        false -> true
+    end.
+
+give_away_does_not_change_heir(_Config) ->
+    Self = self(),
+    {ok, Box} = pobox:start_link([{heir, Self}, {heir_data, first}]),
+    H = erlang:spawn(fun() -> receive X -> Self ! X end, throw("crash") end),
+    true = pobox:give_away(Box, H, second, 5000),
+    true = receive
+        {pobox_transfer, Box, Self, second, give_away} -> true;
+        Other0 -> {false, Other0}
+    after
+        5000 -> timeout
+    end,
+    true = receive
+        {pobox_transfer, Box, H, first, CrashData} when CrashData =/= give_away -> true;
```

Comment:
`pobox_transfer, Box, H, first, CrashData} -> CrashData =/= give_away` ;)

Reply:
There is no error here correct?

Reply:
Changed `CrashData` to `Reason`.